### PR TITLE
Add IPC tests (`umfIpcTest`) to the devdax provider

### PR DESCRIPTION
**Pull request by ldorau**
### Description

Add IPC tests (`umfIpcTest`) to the devdax provider.

### Checklist

- [x] Code compiles without errors locally
- [x] All tests pass locally
- [x] CI workflows execute properly

### CODE REVIEW

**Comment by ldorau** [comment]
See the issue: https://github.com/oneapi-src/unified-memory-framework/issues/846

File: test/provider_devdax_memory.cpp
Comment (vinser52) [inline test/provider_devdax_memory.cpp:19]: Instead of a macro, I suggest extending the `ipcTestParams` to contain a flag that tells if the provider supports free operation.

File: test/ipcFixtures.hpp
Comment (vinser52) [inline test/ipcFixtures.hpp:68]: Why do we need these checks here? In the `provider_devdax_memory.cpp` you can decide whether you would like to instantiate the test or not.

**Review by vinser52 — CHANGES_REQUESTED** [review]

File: test/ipcFixtures.hpp
Comment (ldorau) [reply test/ipcFixtures.hpp:68]: Because if any of the following **environment variables** is not set: `UMF_TESTS_DEVDAX_PATH` or `UMF_TESTS_DEVDAX_SIZE` then the tests should be skipped.

File: test/provider_devdax_memory.cpp
Comment (ldorau) [reply test/provider_devdax_memory.cpp:19]: Done

File: test/provider_devdax_memory.cpp
Comment (vinser52) [inline test/provider_devdax_memory.cpp:358]: Should we also test with other pool implementations, e.g. scalable_pool, jemalloc_pool.

File: test/provider_os_memory.cpp
Comment (vinser52) [inline test/provider_os_memory.cpp:390]: You should also update tests for all providers (L0, etc.) who use it.

File: test/ipcFixtures.hpp
Comment (vinser52) [reply test/ipcFixtures.hpp:65]: I understand your intention to check if the `UMF_TESTS_DEVDAX_PATH` and `UMF_TESTS_DEVDAX_SIZE` environments are set. But I do not think we should check for a particular memory provider here.

My suggestion is the following: instead of always instantiating this test in the `provider_devdax_memory.cpp` on the line 364 and do check here we should not instantiate the `umfIpctests` for devdax provider if the required environment variable were not set. Or you can keep the `ipcProxyPoolTestParamsList` list empty if required environment is not set.

**Review by vinser52 — CHANGES_REQUESTED** [review]

File: test/ipcFixtures.hpp
Comment (ldorau) [reply test/ipcFixtures.hpp:65]: But it would cause multiplication of the code with checking those variables - the same code will have to be placed in each and every provider that uses `ipcFixtures.hpp` - is it really better?

File: test/provider_os_memory.cpp
Comment (ldorau) [reply test/provider_os_memory.cpp:390]: Done

File: test/provider_devdax_memory.cpp
Comment (ldorau) [reply test/provider_devdax_memory.cpp:358]: Done

File: test/ipcFixtures.hpp
Comment (vinser52) [reply test/ipcFixtures.hpp:65]: > the same code will have to be placed in each and every provider that uses ipcFixtures.hpp

Hmm, I did not get your point. This check is specific for DEVDAX provider.

File: test/pools/ipc_jemalloc_devdax.cpp
Comment (vinser52) [inline test/pools/ipc_jemalloc_devdax.cpp:24]: I think this instantiation of the devdax + jemalloc should be done in the `provider_devdax_memory.cpp`. Something like this:

```c++
static std::vector<ipcTestParams> ipcProxyPoolTestParamsList = {
    {umfProxyPoolOps(), nullptr, umfDevDaxMemoryProviderOps(),
    &defaultDevDaxParams, &hostAccessor, false},
#ifdef UMF_JEMALLOC_ENABLED // I do not remember exact macros
    {umfJemallocPoolOps(), nullptr, umfDevDaxMemoryProviderOps(),
     &defaultDevDaxParams, &hostAccessor, false},
#endif
#ifdef UMF_SCALABLE_POOL_ENABLED // I do not remember exact macros
    {umfScalablePoolOps(), nullptr, umfDevDaxMemoryProviderOps(),
     &defaultDevDaxParams, &hostAccessor, false},
#endif
```

I think it is natural to instantiate tests for a particular provider with all available pools in a single palce

Comment (vinser52) [reply test/pools/ipc_jemalloc_devdax.cpp:24]: you can refer to the #860. How I enabled IPC tests for OS provider with scalable pool

File: test/provider_devdax_memory.cpp
Comment (vinser52) [reply test/provider_devdax_memory.cpp:358]: We should enable them here as I mentioned in this comment: https://github.com/oneapi-src/unified-memory-framework/pull/845/files#r1827620658

File: test/ipcFixtures.hpp
Comment (ldorau) [reply test/ipcFixtures.hpp:65]: Yes, I meant for each and every pool (not provider), but if I use https://github.com/oneapi-src/unified-memory-framework/pull/845#discussion_r1827643900 it will be used only once in one file, so it will be OK.

Comment (ldorau) [reply test/ipcFixtures.hpp]: Done, is it OK now?

File: test/provider_devdax_memory.cpp
Comment (ldorau) [reply test/provider_devdax_memory.cpp]: Done, is it OK now?

**Review by vinser52 — APPROVED** [review]

**Review by bratpiorka — APPROVED** [review]

**Comment by bratpiorka** [comment]
@ldorau switched to draft - what do you think - should we merge https://github.com/oneapi-src/unified-memory-framework/pull/761 first? Let's discuss this offline

**Comment by vinser52** [comment]
> @ldorau switched to draft - what do you think - should we merge #761 first? Let's discuss this offline

I have two questions/comments:

1. Why we should postpone merging additional tests?
2. Before merging #761 I suggest to agree on design first.

**Comment by bratpiorka** [comment]
ok we can merge this if CI passes

**Pull request merged**
